### PR TITLE
Convert deprecated listpeer usage to listpeerchannels

### PR DESCRIPTION
**Pull request by ksedgwic**
Construct a "compatibility struct" to convert `listpeerchannels` output into a "virtual" legacy `listpeers` format.

This maintains the semantics of all of the use sites which currently traverse peer, then channel to make decisions and take actions.
    
Tests written using the legacy listpeers format can use the `convert_legacy_listpeers` utility to construct a compatibility struct.
    
The `test_peerjudge_datagatherer` malformed test needed to be malformed differently to achieve the desired effect.

### CODE REVIEW

**Comment by chrisguida** [comment]
Testing on mutinynet

**Review by vincenzopalazzo — APPROVED** [review]
ACK 1cfd8609294f872fe9b034cffab5be3e84e1c4e9

File: Boss/Mod/ConstructedListpeers.cpp [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+			cpeers[id].channels.push_back(c);
+		}
+	}
+	return cpeers;
+}
+
+std::ostream& operator<<(std::ostream& os, Boss::Mod::ConstructedListpeers const& o) {
+	for (auto p : o) {
+		os << p.first << ':'
+                   << "connected: " << p.second.connected
```
Comment (chrisguida) [inline Boss/Mod/ConstructedListpeers.cpp:37]: curious what this `second` does

Comment (vincenzopalazzo) [reply Boss/Mod/ConstructedListpeers.cpp:37]: access the `(key, vale)` in c++ is `(first, second)` 

I bet that ConstructedListpeers is a `map`

Comment (chrisguida) [reply Boss/Mod/ConstructedListpeers.cpp:37]: Ah cool. How can I verify this? Struggling to find the definition of `ConstructedListpeers`

Comment (vincenzopalazzo) [reply Boss/Mod/ConstructedListpeers.cpp:37]: What I do usually is `git grep -n "ConstructedListpeers"` and then search inside the result

In this case, I found the following line `Boss/Mod/ConstructedListpeers.hpp:24:typedef std::map<Ln::NodeId, ConstructedListpeer> ConstructedListpeers;`

Comment (chrisguida) [reply Boss/Mod/ConstructedListpeers.cpp:37]: Ahh, thanks!! Still getting familiar with how C++ does things...

**Comment by ksedgwic** [comment]
rebased and merging

**Pull request merged**